lib/Service/FeedServiceV2.php

```diff
@@ -342,4 +343,21 @@ public function fetchAll(): void
             $this->fetch($feed);
         }
     }
+
+    /**
+     * Mark a feed as read
+     *
+     * @param string   $userId    Feed owner
+     * @param int      $id        Feed ID
+     * @param int|null $maxItemID Highest item ID to mark as read
+     *
+     * @throws ServiceConflictException
+     * @throws ServiceNotFoundException
+     */
+    public function read(string $userId, int $id, ?int $maxItemID = null): void
+    {
+        $feed = $this->find($userId, $id);
+
+        $this->mapper->read($userId, $feed->getId(), $maxItemID);
```

Comment:
why use `find()` with the user id and feed id, to just use the feed id from the returned entity?

Reply:
As a check if the ID exists

Reply:
The database shouldn't care if the `feeds.id = :feedId` is a not existing id.
Do need it for the exception? Otherwise I would avoid requesting the full feed metadata just for fun.

Reply:
It doesn't care, but I think we should throw meaningful errors when a developer is doing something that won't work. Otherwise we're allowing someone to introduce a bug that would be near impossible to catch.